policy/predicate/predicate.go

```diff
@@ -26,5 +26,5 @@ type Predicate interface {
 
 	// Evaluate determines if the predicate is satisfied. It also returns an
 	// optional string providing details about the evaluation result.
-	Evaluate(ctx context.Context, prctx pull.Context) (bool, string, error)
+	Evaluate(ctx context.Context, prctx pull.Context) (bool, string, *common.PredicateInfo, error)
```

Comment:
It's unfortunate that this now returns 4 values. I think it would be best if we could combine the existing string description with the new `PredicateInfo` value. Maybe make it a field of the struct or add a function to render the struct into a text description similar to how we render it in to HTML?

Reply:
I put `desc` as a field in `PredicateInfo`, so now we are returning 3 again.